dan_layer/consensus/src/hotstuff/on_receive_proposal.rs

```diff
@@ -133,30 +166,48 @@ where TConsensusSpec: ConsensusSpec
             Ok::<_, HotStuffError>(())
         })?;
 
+        if self.block_has_missing_transaction(&local_committee, &block).await? {
+            Ok(())
```

Comment:
It might be useful for testing to return an enum saying which decision we took

```suggestion
            Ok(LocalProposalResult::MissingTransactions)
```

Reply:
In the test we don't call the function directly, it's inside a loop over tokio::select, so the return value doesn't bring any value for testing. But if the missing transactions is triggered we can check if the node sent the `RequestMissingTransactions`. That should be enough for testing.